components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,352 @@
+
+	return options
+end
+
+function PrizePool:_readPrizes(args)
+	local prizes = {}
+
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		local fieldName = prizeDatum.header
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, prizeValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				local data = prizeDatum.headerParse(self, prizeValue, args, index)
+				self:addPrize(prizeEnum, index, data, prizes)
+			end
+		end
+	end
+
+	return prizes
+end
+
+function PrizePool:_readPrizeRewards(args)
+	local rewards = {}
+
+	for prizeEnum in pairs(prizeTypes) do
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		local fieldName = prizeDatum.row
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, rewardValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				rewards[prizeEnum .. index] = prizeDatum.rowParse(self, rewardValue, args, index)
+			end
+		end
+	end
+
+	return rewards
+end
+
+function PrizePool:_readSpecialData(args)
+	local data = {}
+
+	for enum in pairs(specialDataTypes) do
+		local type = specialData[specialDataTypes[enum]]
+		local fieldName = type.field
+		if args[fieldName] then
+			data[enum] = type.parse(self, args[fieldName], args)
+		end
+	end
+
+	return data
+end
+
+function PrizePool:_readPlacements(args)
+	local placements = {}
+
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		local placementInput = Json.parseIfString(args[placementIndex])
+		if not placementInput then
+			break
+		end
+
+		local placement = {}
+		-- Parse place, explicit
+		if placementInput.place then
+			local places = Table.mapValues(mw.text.split(placementInput.place, '-'), tonumber)
+			placement.placeStart = places[1]
+			placement.placeEnd = places[2] or places[1]
+		end
+
+		-- Parse prizes
+		placement.prizes = self:_readPrizeRewards(placementInput)
+
+		-- Parse opponents in the placement
+		placement.opponents = {}
+		for opponentIndex = 1, math.huge do
+			local opponentInput = Json.parseIfString(placementInput[opponentIndex])
+			local opponent = {opponentData = {}, prizes = {}, data = {}}
+			if not opponentInput then
+				-- If given a range of opponents, add them all, even if they're missing from the input
+				if not placementInput.place or placement.placeStart + opponentIndex > placement.placeEnd + 1 then
+					break
+				else
+					opponent.opponentData = Opponent.tbd()
+				end
+			else
+				-- Parse Opponent Data
+				opponentInput.type = opponentInput.type or self.opponentType
+				opponent.opponentData = self:_parseOpponentArgs(opponentInput, opponentInput.date)
+
+				-- Parse special prizes for this opponent
+				opponent.prizes = self:_readPrizeRewards(opponentInput)
+
+				-- Parse additional data (groupscore, last opponent etc)
+				opponent.data = self:_readSpecialData(opponentInput)
+
+				-- Set the date
+				opponent.date = opponentInput.date
```

Comment:
```suggestion
				-- Set the date
				opponent.date = opponentInput.date or self.date

				opponent.opponentData = self:_parseOpponentArgs(opponentInput, opponent.date)

				-- Parse special prizes for this opponent
				opponent.prizes = self:_readPrizeRewards(opponentInput)

				-- Parse additional data (groupscore, last opponent etc)
				opponent.data = self:_readSpecialData(opponentInput)
```

move the date stuff before the `_parseOpponentArgs` call, use self.date as fallback and then pass `opponent.date` to `_parseOpponentArgs`

Reply:
Changed